flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/optimize/program/FlinkChangelogModeInferenceProgram.scala

```diff
@@ -310,18 +310,26 @@ class FlinkChangelogModeInferenceProgram extends FlinkOptimizeProgram[StreamOpti
           .newBuilder()
           .addContainedKind(ModifyKind.INSERT)
 
-        // All aggregates are computed over the same window and order by is supported for only 1 field
-        val orderKeyIndex =
-          over.logicWindow.groups.get(0).orderKeys.getFieldCollations.get(0).getFieldIndex
-        val orderKeyType = over.logicWindow.getRowType.getFieldList.get(orderKeyIndex).getType
         if (
-          !FlinkTypeFactory.isRowtimeIndicatorType(orderKeyType)
-          && !FlinkTypeFactory.isProctimeIndicatorType(orderKeyType)
+          !over.logicWindow.groups.isEmpty && !over.logicWindow.groups
```

Comment:
can we extract `over.logicWindow.groups` into a var?

Reply:
Done